sql/core/src/test/scala/org/apache/spark/sql/sources/v2/DataSourceV2SQLSuite.scala

```diff
@@ -174,26 +180,17 @@ class DataSourceV2SQLSuite extends QueryTest with SharedSQLContext with BeforeAn
   }
 
   test("CreateTableAsSelect: use v2 plan and session catalog when provider is v2") {
-    spark.sql(s"CREATE TABLE table_name USING $orc2 AS SELECT id, data FROM source")
-
-    val testCatalog = spark.catalog("session").asTableCatalog
-    val table = testCatalog.loadTable(Identifier.of(Array(), "table_name"))
-
-    assert(table.name == "session.table_name")
-    assert(table.partitioning.isEmpty)
-    assert(table.properties == Map("provider" -> orc2).asJava)
-    assert(table.schema == new StructType()
-        .add("id", LongType, nullable = false)
-        .add("data", StringType))
-
-    val rdd = spark.sparkContext.parallelize(table.asInstanceOf[InMemoryTable].rows)
-    checkAnswer(spark.internalCreateDataFrame(rdd, table.schema), spark.table("source"))
+    // TODO: support write table from v2 session catalog.
```

Comment:
Note that, this is never supported. The test can pass before because we set a fake session catalog which returns writable tables.

Reply:
To clarify, this doesn't work because the ORC source is broken.

The test case is still valid, so I don't see a reason why we should get rid of it. I also think that it is a valid use to override the v2 session catalog implementation, which this relies on for tests.